### PR TITLE
EUI-1620 - date display format

### PR DESCRIPTION
**Pull request by andywilkinshmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-1620

### Change description ###
Implement date formatting for display of dates.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```
Backwards compatibility maintained, as evidenced by the tests. However there was a mostly non-functional date parameter on the date pipe which now works as expected. Any code which called the date pipe passing a date format will now get most likely get a different result. However no service code currently does this.

### CODE REVIEW

**Review by adnanakgun — CHANGES_REQUESTED** [review]
I believe you are missing the updated yarn.lock file. Everything else seems fine.

File: src/shared/components/palette/utils/date.pipe.ts [hunk 1]
```diff
@@ -24,20 +33,35 @@ export class DatePipe implements PipeTransform {
...
-        resultDate += this.getHour(offsetDate.getHours().toString()) + ':';
-        resultDate += this.pad(offsetDate.getMinutes()) + ':';
-        resultDate += this.pad(offsetDate.getSeconds()) + ' ';
-        resultDate += (this.toInt(offsetDate.getHours().toString()) >= 12) ? 'PM' : 'AM' ;
+      // 'short' format is meaningful to formatDate, but not the same meaning as in the unit tests
+      if (this.formatTrans && format && format !== 'short') {
+        // support for java style formatting strings for dates
+        format = this.translateDateFormat(format);
+        // we specify UTC because we've already done timezone offsetting if the zone === local
+        resultDate = formatDate(offsetDate, format, 'en-UK', null)
```
Comment (Daniel-Lam) [inline src/shared/components/palette/utils/date.pipe.ts:41]: I thought the timezone was `en-GB`? Might be inconsequential but thought I'd check

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@andywilkinshmcts - just the one comment on timezone identifier. Also, do you have tests that check the correct behaviour when the time is transitioning from GMT to BST (from 00:59 GMT going forward to 02:00 BST) on the last Sunday in March and vice versa (from 01:59 BST going back to 01:00 GMT) on the last Sunday in October?

**Review by Daniel-Lam — APPROVED** [review]
@andywilkinshmcts - nice work on the additional unit tests 👍, approved.

**Pull request merged**